src/internal/connector/onedrive/item.go

```diff
@@ -250,11 +249,12 @@ func filterUserPermissions(ctx context.Context, perms []models.Permissionable) [
 		} else if gv2.GetGroup() != nil {
 			entityID = ptr.Val(gv2.GetGroup().GetId())
 		} else {
-			// TODO Add application permissions when adding permissions for SharePoint
-			// https://devblogs.microsoft.com/microsoft365dev/controlling-app-access-on-specific-sharepoint-site-collections/
 			logm := logger.Ctx(ctx)
+
 			if gv2.GetApplication() != nil {
-				logm.With("application_id", ptr.Val(gv2.GetApplication().GetId()))
+				entityID = ptr.Val(gv2.GetApplication().GetId())
```

Comment:
This is not required for restore as of now, but I was just wondering if we should store what kind of permissions we are storing(user or group or device etc) just in case we need that information later on for filtering permissions to restore.

Reply:
@meain Can you please make a tech-debt ticket to add that?  We can follow up from there as needed.